### PR TITLE
mysql@5.7 5.7.40

### PR DESCRIPTION
**Pull request by chenrui333**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by carlocab** [comment]
Linux failing with
`  /usr/bin/ld: CMakeFiles/innodb_engine.dir/src/innodb_engine.c.o:(.bss.ib_cb_cfg_trx_level+0x0): multiple definition of `ib_cb_cfg_trx_level'; CMakeFiles/innodb_engine.dir/src/innodb_config.c.o:(.bss.ib_cb_cfg_trx_level+0x0): first defined here`

**Comment by Bo98** [comment]
Can either use `-fcommon` or use https://github.com/oracle/solaris-userland/blob/master/components/mysql-5-7/patches/gcc10.patch.

File: Formula/mysql@5.7.rb [hunk 1]
```diff
@@ -44,6 +45,9 @@ def install
       # against `_ZN17Gcs_debug_options12m_debug_noneB5cxx11E' can not be used when making
       # a shared object; recompile with -fPIC
       ENV.append_to_cflags "-fPIC"
+      # Work around failure from GCC 10+ using default of `-fno-common
+      # innodb_engine.c.o:(.bss.ib_cb_cfg_trx_level+0x0): multiple definition of `ib_cb_cfg_trx_level';
+      ENV.append_to_cflags "-fcommon"
```
Comment (SMillerDev) [inline Formula/mysql@5.7.rb:50]: When can this be removed?

Comment (Bo98) [reply Formula/mysql@5.7.rb:50]: Given Oracle's own OS still carry around a GCC 10 patch for MySQL 5.7, I wouldn't be surprised if it was never.

MySQL 5.7 is EOL in October 2023.

Comment (SMillerDev) [reply Formula/mysql@5.7.rb:50]: Should probably add a deprecation then. And a reference to the upstream patch

**Pull request closed without merging**